### PR TITLE
Add Nano support

### PR DESCRIPTION
**Pull request by roosmaa**
This PR adds new Nano specific message types and adds nanoaddr_prefix to CoinType.

I based my work on v5.11.0, I'll get to rebasing everything on the latest relase tomorrow.

### CODE REVIEW

**Comment by roosmaa** [comment]
Rebased on latest master.

File: messages-nano.proto [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    repeated uint32 address_n = 1;  // BIP-32 path to derive the key from master node
+    optional string coin_name = 2 [default='Nano'];
+    optional ParentBlock parent_block = 3;
+    optional bytes link_hash = 4;
+    optional string link_recipient = 5;
+    repeated uint32 link_recipient_n = 6;         // BIP-32 path to derive address for recipient
+    optional OutputAddressType address_type = 7;  // output address type
+    optional ExchangeType exchange_type = 8;      // exchange type data
+    required string representative = 9;
+    required bytes balance = 10;  // uint128 (big-endian encoded)
```
Comment (keepkeyjon) [inline messages-nano.proto]: Please don't use `required`. That constraint needs to be enforced by firmware instead of by the protobuf library. That affords us more flexibility if we need to change these message fields later.

Comment (roosmaa) [reply messages-nano.proto]: Got it. Will add explicit checks for those fields in the firmware and also integration tests to check that empty values are caught.

Comment (keepkeyjon) [reply messages-nano.proto]: I was just in the middle of doing that :) see: https://github.com/keepkey/device-protocol/pull/21/commits/98bcbf72eee4d4f65634c8f87403ed1552e8da89

Comment (roosmaa) [reply messages-nano.proto]: Other PRs have been updated to handle the `required` -> `optional` changes.
